Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm

```diff
@@ -643,19 +643,44 @@ void displayLinkFired(PlatformDisplayID displayID, DisplayUpdate, bool wantsFull
         animationStack->clear(node.protectedLayer().get());
 }
 
-void RemoteLayerTreeEventDispatcher::registerTimelineIfNecessary(WebCore::ProcessIdentifier processIdentifier, Seconds originTime, MonotonicTime now)
+void RemoteLayerTreeEventDispatcher::registerTimelinesIfNecessary(WebCore::ProcessIdentifier processIdentifier, const HashSet<Ref<WebCore::AcceleratedTimeline>>& timelineRepresentations, MonotonicTime now)
 {
     assertIsHeld(m_animationLock);
-    if (m_timelines.find(processIdentifier) == m_timelines.end())
-        m_timelines.set(processIdentifier, RemoteAnimationTimeline::create(originTime, now));
+    for (auto& timelineRepresentation : timelineRepresentations) {
+        auto foundExistingTimeline = false;
+
+        auto it = m_timelines.find(processIdentifier);
+        if (it != m_timelines.end()) {
+            auto& existingMonotonicTimelines = it->value;
+            for (auto& existingTimeline : existingMonotonicTimelines) {
+                if (existingTimeline->identifier() == timelineRepresentation->identifier()) {
+                    existingTimeline->setOriginTime(timelineRepresentation->originTime());
```

Comment:
Do we expect the origin time for a given timeline to be mutable?

Reply:
Actually, it can only be set at construction time for a `DocumentTimeline` in the web process, so cannot be updated on the remote side. I'll remove this code and the `RemoteAnimationTimeline::setOriginTime()` method.